### PR TITLE
LP-206 create the partnership POST endpoint

### PR DESCRIPTION
**Pull request by markpit**
### JIRA link
https://companieshouse.atlassian.net/browse/LP-206

### Change description
Update the POST endpoint for /partnership to receive a request body and store it in mongo.
Returns the created user id.

### Work checklist

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__
### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/controller/PartnershipController.java [hunk 1]
```diff
@@ -1,30 +1,61 @@
...
+
+        try {
+            ApiLogger.infoContext(requestId, "Calling service to create Partnership Submission", logMap);
+
+            var submissionId = limitedPartnershipService.createLimitedPartnership(limitedPartnershipSubmissionDto, requestId, userId);
 
-        ApiLogger.debug("createPartnership");
+            var location = URI.create(String.format(URL_GET_PARTNERSHIP, transactionId, submissionId));
+            var response = new LimitedPartnershipSubmissionCreatedResponseDto();
+            response.setId(submissionId);
```
[the remark this answers is not in the document]
Comment (markpit) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/controller/PartnershipController.java]: Updated, Intellij then recommended converting the class to a Record, so have done that as well.

File: src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/utils/Constants.java [hunk 2]
```diff
@@ -6,7 +6,12 @@ private Constants() { }
...
     // Request header names
     public static final String ERIC_REQUEST_ID_KEY = "X-Request-Id";
+    public static final String ERIC_IDENTITY = "ERIC-identity";
 
     // URL path parameters
     public static final String URL_PARAM_TRANSACTION_ID = "transaction_id";
+
+    // URLs
+    public static final String ENDPOINT_PARTNERSHIP = "/transactions/{" + URL_PARAM_TRANSACTION_ID + "}/limited-partnership/partnership";
+    public static final String URL_GET_PARTNERSHIP = "/transactions/%s/limited-partnership/partnership/%s";
```
[the remark this answers is not in the document]
Comment (markpit) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/utils/Constants.java]: Moved these into the controller

**Review by dlloyd1-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/model/dto/LimitedPartnershipSubmissionCreatedResponseDto.java [hunk 3]
```diff
@@ -2,16 +2,4 @@
...
-
-    public String getId() {
-        return id;
-    }
-
-    public void setId(String id) {
-        this.id = id;
-    }
-}
+public record LimitedPartnershipSubmissionCreatedResponseDto(@JsonProperty("id") String id) {}
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/model/dto/LimitedPartnershipSubmissionCreatedResponseDto.java:5]: New feature! 😄

**Review by mwestacott — APPROVED** [review]

**Pull request merged**